definitions/npm/react-redux_v5.x.x/flow_v0.68.0-/test_Provider.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+import { describe, it } from 'flow-typed-test';
+
+import React from "react";
+import { Provider, createProvider } from "react-redux";
+
+describe('Provider', () => {
+  it('should give an error when the store is missing', () => {
+    // $ExpectError
+    <Provider />;
+
+    // Also for custom providers
+    const CustomProvider: Provider<*, *, *> = createProvider("ikea");
+
+    // $ExpectError
+    <CustomProvider />;
+  });
+});
+
+describe('Custom Store (eg for ThunkActions)', () => {
+
+  // This represents a common typing for Thunk Actions.
+
+  type Action = { type: 'SOME_ACTION' };
+  type State = { state: string };
+
+  // ReduxStore should be imported from 'redux' but we can't do this with this
+  // test environment, so let's copy them once again...
+  declare type Redux$DispatchAPI<A> = (action: A) => A;
+  declare type Redux$Dispatch<A: { type: $Subtype<string> }> = Redux$DispatchAPI<A>;
+  declare type Redux$Reducer<S, A> = (state: S | void, action: A) => S;
+  declare type Redux$Store<S, A, D = Redux$Dispatch<A>> = {
+    dispatch: D;
+    getState(): S;
+    subscribe(listener: () => void): () => void;
+    replaceReducer(nextReducer: Redux$Reducer<S, A>): void
```

Comment:
for some reason, this was failing in Flow < 0.68... that's why I duplicated the folder.
Here is the error I got, in case you'd have an idea:
```
ERROR: react-redux_v5.x.x/flow_v0.63.0-
 * react-redux_v5.x.x/flow_v0.63.0- (flow-v0.67.1): Unexpected Flow errors(2):
   Error -------------------------------------------------------------------------------------------- test_Provider.js:52:5
   
   Cannot create `Provider` element because `State` [1] is incompatible with undefined [2] in the return value of the first
   argument of property `store.replaceReducer`.
   
      test_Provider.js:52:5
      52|     <Provider store={store} />;
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
   
   References:
      test_Provider.js:48:28
      48|   type Store = Redux$Store<State, Action, Dispatch>;
                                     ^^^^^ [1]
      test_Provider.js:31:50
      31|   declare type Redux$Reducer<S, A> = (state: S | void, action: A) => S;
                                                           ^^^^ [2]

```